openbb_platform/obbject_extensions/charting/openbb_charting/query_params.py

```diff
@@ -103,3 +218,544 @@ class TechnicalConesChartQueryParams(QueryParams):
+        default=14,
+        description="Window length for the RSI, by default is 14.",
+    )
+    scalar: float = Field(
+        default=100,
+        description="Scalar to multiply the RSI by, default is 100.",
+    )
+    drift: int = Field(
+        default=1,
+        description="Drift value for the RSI, by default is 1.",
+    )
+
+
+class StochIndicatorsQueryParams(IndicatorsQueryParams):
+    """Stochastic Oscillator Indicators Query Params."""
+
+    fast_k: int = Field(
+        default=14,
+        description="The fast K period, by default 14.",
+    )
+    slow_d: int = Field(
+        default=3,
+        description="The slow D period, by default 3.",
+    )
+    slow_k: int = Field(
+        default=3,
+        description="The slow K period, by default 3.",
+    )
+
+
+class FibIndicatorsQueryParams(IndicatorsQueryParams):
+    """Fibonacci Retracement Indicators Query Params."""
+
+    period: int = Field(
+        default=120,
+        description="The period to calculate the Fibonacci Retracement, by default 120.",
+    )
+    start_date: Optional[str] = Field(
+        default=None,
+        description="The start date for the Fibonacci Retracement.",
+    )
+    end_date: Optional[str] = Field(
+        default=None,
+        description="The end date for the Fibonacci Retracement.",
+    )
+
+
+class ClenowIndicatorsQueryParams(IndicatorsQueryParams):
+    """Clenow Volatility Adjusted Momentum Indicators Query Params."""
+
+    period: int = Field(
+        default=90,
+        description="The number of periods for the momentum, by default 90.",
+    )
+
+
+class DemarkIndicatorsQueryParams(IndicatorsQueryParams):
+    """Demark Indicators Query Params."""
+
+    show_all: bool = Field(
+        default=False,
+        description="Show 1 - 13. If set to False, show 6 - 9.",
+    )
+    offset: int = Field(
+        default=0,
+        description="Number of periods to offset the result, by default is 0.",
+    )
+
+
+class IchimokuIndicatorsQueryParams(IndicatorsQueryParams):
+    """Ichimoku Cloud Indicators Query Params."""
+
+    conversion: int = Field(
+        default=9,
+        description="The conversion line period, by default 9.",
+    )
+    base: int = Field(
+        default=26,
+        description="The base line period, by default 26.",
+    )
+    lagging: int = Field(
+        default=52,
+        description="The lagging line period, by default 52.",
+    )
+    offset: int = Field(
+        default=26,
+        description="The offset period, by default 26.",
+    )
+    lookahead: bool = Field(
+        default=False,
+        description="Drops the Chikou Span Column to prevent potential data leak",
+    )
+
+
+class SRLinesIndicatorsQueryParams(IndicatorsQueryParams):
+    """Support and Resistance Lines Indicators Query Params."""
+
+    show: bool = Field(
+        default=True,
+        description="Show the support and resistance lines.",
+    )
+
+
+class IndicatorsParams(QueryParams):
```

Comment:
Indicators should be validated against `openbb_charting.core.plotly_ta.data_classes.TAIndicator.name`: which contains the list of available indicators.